File: parent/monitoring/net.bluemind.monitoring.api/src/net/bluemind/monitoring/api/Command.java
/* BEGIN LICENSE
  * Copyright © Blue Mind SAS, 2012-2016
  *
  * This file is part of Blue Mind. Blue Mind is a messaging and collaborative
  * solution.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of either the GNU Affero General Public License as
  * published by the Free Software Foundation (version 3 of the License).
  *
  *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  *
  * See LICENSE.txt
  * END LICENSE
  */
package net.bluemind.monitoring.api;

import java.util.List;

import net.bluemind.core.api.BMApi;

/**
 * A command is used to execute Shell scripts and contains the results of its
 * execution (raw and in a list).
 */
@BMApi(version = "3")
public class Command {

	/**
	 * The Shell script to be executed.
	 */
	public String commandLine;

	/**
	 * The raw (non-formatted) result of the command's execution.
	 */
	public String rawData;

	/**
	 * Formatted data in a list.
	 */
	public List<FetchedData> dataList;

	public Command() {
	}

	/**
	 * Creates a new {@link Command} with a given command line.
	 * 
	 * @param commandLine the command line to be executed
	 */
	public Command(String commandLine) {
		this.commandLine = commandLine;
		this.rawData = new String();
	}

	/**
	 * Be careful, {@link #dataList} can be modified unlike {@link #rawData}!
	 * 
	 * @return <code>true</code> if the formatted data list is empty or
	 *         <code>null</code>, <code>false</code> otherwise
	 */
	public boolean hasDataList() {
		return (this.dataList != null && !this.dataList.isEmpty());
	}
}

File: ui/closures/net.bluemind.commons.closure/src/main/javascript/cmd/command.js
/**
 * BEGIN LICENSE
 * Copyright © Blue Mind SAS, 2012-2016
 *
 * This file is part of BlueMind. BlueMind is a messaging and collaborative
 * solution.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of either the GNU Affero General Public License as
 * published by the Free Software Foundation (version 3 of the License).
 *
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * See LICENSE.txt
 * END LICENSE
 */


/**
 * @fileoverview
 * 
 * Command is an abstract class extending RPC Command.
 * It add additionals parameters like service and method to match the bluemind
 * request structure.
 * 
 */


goog.provide('bluemind.cmd.Command');

goog.require('goog.Uri.QueryData');
goog.require('relief.rpc.Command');
goog.require('goog.json.Serializer');


/**
 * Command for requesting bluemind server.
 * @param {Function} onSuccess Called on a successful response.
 * @param {Function} onFailure Called if the request fails.
 * @param {string} commandID The ID for this command.
 * @param {string} url The URL to request.
 * @param {string=} opt_service Requested servlet.
 * @param {string=} opt_action The method to execute on the server side.
 * @param {number=} opt_maxRetries The number of retries.
 * @constructor
 * @extends {relief.rpc.Command}
 */
bluemind.cmd.Command = function(onSuccess, onFailure, commandID, url, opt_service,
  opt_action, opt_maxRetries) {
  this.data = new goog.Uri.QueryData();
  if (opt_service) this.data.add('service', opt_service);
  if (opt_action) this.data.add('method', opt_action);
  this.serializer = new goog.json.Serializer();
  goog.base(this, onSuccess, onFailure, commandID, url, 'POST',
    opt_maxRetries);
};
goog.inherits(bluemind.cmd.Command, relief.rpc.Command);

/**
 * Post content
 * @type {goog.Uri.QueryData}
 * @protected
 */
bluemind.cmd.Command.prototype.data;

/**
 * The JSON serializer used to serialize values.
 *
 * @type {goog.json.Serializer}
 * @protected
 */
bluemind.cmd.Command.prototype.serializer = null;

/** @override */
bluemind.cmd.Command.prototype.getData = function() {
  return this.data.toString();
};
